src/owl/core/owl_tensor.ml

```diff
@@ -0,0 +1,144 @@
+(*
+ * OWL - an OCaml numerical library for scientific computing
+ * Copyright (c) 2016-2017 Liang Wang <liang.wang@cl.cam.ac.uk>
+ *)
+
+(** Tensor: tensor implementation in Owl *)
+
+open Bigarray
+open Owl_types
+open Owl_dense_common_types
+
+external owl_float32_tensor_maxpool_spatial : ('a, 'b) owl_arr -> ('a, 'b) owl_arr -> int -> int -> int -> int -> int -> int -> int -> int -> int -> int -> int -> int -> int -> unit =
```

Comment:
`Tensor` this term has been abused recently for obvious reasons. In Owl, we stick to ndarray which simply means n-dimensional array. I suggest you to change all these (including file names) from tensor to ndarray, so these will be consistent with the naming scheme used in Owl.

Reply:
The `Tensor` is taken from Eigen's naming format. Sure, I'll update them.

Reply:
I understand that, because the Tensor implementation in Eigen is provided by Google, now we are removing our dependency on Eigen, so it is a good time to change it back to ndarray, avoid future confusion.